### PR TITLE
Print warn messages as opposed to throwing exceptions in some OOC paths.

### PR DESCRIPTION
**Pull request by dlogothetis**
Not necessary to throw exceptions here and potentially confuse users, just print some warning so that we're aware of it.

### CODE REVIEW

**Comment by dlogothetis** [comment]
Test done:
- snapshot tests

File: giraph-core/src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java [hunk 1]
```diff
@@ -770,16 +787,17 @@ private boolean refineCoefficient(int coefIndex, double lowerBound,
      * @param coefficient Array of coefficients
      * @param validColumnIndices List of valid columns
      * @param mlr {@link OLSMultipleLinearRegression} instance.
+     * @return True if the result is valid, false otherwise.
      * @throws Exception
```
Comment (heslami) [inline giraph-core/src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java]: You can get rid of the `Exception` in the comments.

File: giraph-core/src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java [hunk 2]
```diff
@@ -706,12 +721,12 @@ public void addRecord(long memUsed, long edges, long vertices,
      * @param upperBound Upper bound
      * @param xValues double[][] matrix with data samples
      * @param yValues double[] matrix with y samples
-     * @return True if coefficients were out-of-range
+     * @return True if coefficients were out-of-range, false otherwise. A null value means the
+     *         regression result was invalid and the result of this method is invalid too.
      * @throws Exception
```
Comment (heslami) [inline giraph-core/src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java]: You can get rid of the `Exception` in the comments.

File: giraph-core/src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java [hunk 3]
```diff
@@ -653,8 +657,19 @@ public void addRecord(long memUsed, long edges, long vertices,
...
+            if (result == null) { // invalid regression result
+              return;  // finally-block will release lock
+            }
+            changed = result;
+
+            result = refineCoefficient(3, 0, 2, xValues, yValues);
+            if (result == null) { // invalid regression result
+              return;  // finally-block will release lock
+            }
+            changed |= result;
```
Comment (heslami) [inline giraph-core/src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java:671]: You can now get rid of the `catch` block too.

**Review by heslami — CHANGES_REQUESTED** [review]
Some style-check errors.

[ERROR] src/main/java/org/apache/giraph/ooc/OutOfCoreIOStatistics.java[57:3] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/org/apache/giraph/ooc/OutOfCoreIOStatistics.java[86:3] (javadoc) JavadocVariable: Missing a Javadoc comment.
[ERROR] src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java[612] (sizes) LineLength: Line is longer than 80 characters (found 96).
[ERROR] src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java[721] (sizes) LineLength: Line is longer than 80 characters (found 94).
[ERROR] src/main/java/org/apache/giraph/ooc/policy/MemoryEstimatorOracle.java[722] (sizes) LineLength: Line is longer than 80 characters (found 90).

**Review by heslami — APPROVED** [review]
I'll commit this

**Pull request closed without merging**